### PR TITLE
Add Jekyll to site

### PR DESCRIPTION
**Pull request by derrekchow**
**Note: we could merge this after the Summit**

## What
Jekyll is a light framework that allows you to extract repeated html to individual files and separate an element's data from its html. I've added Jekyll to the site, note that its not too big of a change. It mostly means I've changed our folder structure and extracted repeated parts of pages to individual files.

## Why
- To make it easier and quicker to make small changes
- Help make development in general easier by **grouping files by their use instead of by page**
- Reduce code size and repeated elements
- Avoiding having to maintain consistency across pages with duplicate elements (like navbar and footer)

## Changes
- No more `node_modules/` and package.json -> replaced with `Gemfile` and `_config.yml`
- No more Gulp, just run `jekyll serve` - files are auto compressed and SASS auto generated with hot reloading
- Less code per page

Below are the folders than contain the majority of the site's content and their use:

### _data/
Speakers, team members, and nav items are represented as lists in `_data/`. That way don't have to repeat or edit html elements when modifying these entities, we just change the actual data and render each element like so: 
```
{% for member in site.data.members %}
<a name="{{member.name}}" body="{{member.body}}" image="/assets/images/teamphotos/{{member.image}}" >
    <div>
        <img src="/assets/images/teamphotos/{{member.image}}" alt="{{member.name}}" />
        <h3 class="text-dark mt-3">{{member.name}}</h3>
    </div>
</a>
{% endfor %}
```

### _includes/
HTML elements that persist across the entire site like the navbar, footer, and head tags are kept as individual files in `_includes/` and "included" into `_layouts/main.html`. That way, editing every footer across the site means editing just a single file: `/_includes/footer.html`.
```
---
layout: compress
---
<!DOCTYPE html>
<html>
    <head>{% include head.html %}</head>
    <body>
        {% include nav.html %}
        <main>{{ content }}</main>
        {% include footer.html %}
        {% include scripts.html %}
    </body>
</html>
```

### pages/
As shown in the section above, `{{ content }}` is the page specific html which remains the same as before. For example in `pages/coalition.html`, the file is prefixed with:

```
---
title: Impact Labs - Coalition
permalink: /coalition
---
```
This means when the user goes to `impactlabs.io/coalition` then the content in `pages/coalition.html` will be inserted in the place of `{{ content }}` in `_layouts/main.html`.

## Setup
1.  Install [Jekyll](https://jekyllrb.com/docs/installation/)
2.  Run `bundle install`
3.  Run `jekyll serve`

## Layout
```
.
├── _config.yml
├── _data
├── _includes -> Page elements that persist across the site (navbar, footer, head), inserted into main.html in /layouts
├── _layouts -> All pages are rendered through here
├── _sass -> main.scss (edit this and not /assets/scss/styles.scss)
├── _site -> Auto generated by Jekyll (ignore)
├── assets -> Static assets (images, CSS and JS files)
└── pages -> Body of each page, to be inserted in /layouts/main.html
```

## TODO
- Fully setup [jekyll-seo-tag](https://github.com/jekyll/jekyll-seo-tag)
- Make sure `Jekyll` works for everyone
- Decide together if this is preferable to the old site setup

### CODE REVIEW

File: _data/fellowship_speakers.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# image is located at `/assets/images/speakers`
+# note: the first speaker must have active: true
+
+- name: Amy Nelson
```
Comment (Holben888) [inline _data/fellowship_speakers.yaml:4]: Love me some YAML 😍

File: assets/css/styles.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (Holben888) [inline assets/css/styles.scss:1]: Hm, stray empty frontmatter? Seems to break the build for me

File: assets/js/script.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+$(document).ready(function() {
```
Comment (Holben888) [inline assets/js/script.js:1]: Future note: maybe rewrite this without the JQuery? Sounds like the latest bootstrap dropped the JQuery dependency 👍

File: pages/coalition.html [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		<div class="col-md-8 col-9">
+			<h3 style="color: black">Individual company interview</h3>
+			<p>The last step: you'll select your top 5 preferences of the companies within the coalition and interview with them. Salary, start date, and role are all finalized at this time.</p>
+		</div>
+		</div>
+	</div>
+</section>
+
+<section>
+	<img alt="Plus" class="accent d-none d-md-inline" style="height: 130px; width: 130px; position:relative; float:right; bottom:100px; right: 100px;" src="/assets/images/accents/plus.svg" />
```
Comment (Holben888) [inline pages/coalition.html:94]: Could possibly inline these SVGs now that we have liquid syntax! [Found a slim plugin for this here](https://github.com/sdumetz/jekyll-inline-svg)

File: pages/team.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
+---
+title: Impact Labs - Team
+permalink: /team
+---
+
+<!-- Temp styling in file, will be added in SCSS files -->
+<style>
```
Comment (Holben888) [inline pages/team.html:7]: Future note: Could probably chuck this in the scss file now 😆

**Review by Holben888 — APPROVED** [review]
Really hyped for this change overall! Noticing some build issues and had a few nits; otherwise lgtm 👍

File: pages/coalition.html (hunk 4 above)
Comment (derrekchow) [reply pages/coalition.html:94]: Agreed, theres all sorts of Jekyll plugins we can add if we want!

File: assets/js/script.js (hunk 3 above)
Comment (derrekchow) [reply assets/js/script.js:1]: Yeah, although I don't think it greatly affects the load time.

Comment (derrekchow) [reply assets/js/script.js:1]: I'd like to get rid of AOS (scroll to library) if possible

File: pages/team.html (hunk 5 above)
Comment (derrekchow) [reply pages/team.html]: yup sure, might have to clean up the scss eventually

**Comment by Holben888** [comment]
Closing this PR since it can't cleanly sit on top of master. I'm making a new branch called `main` which will replace the `master` shortly. It's a worthwhile change from a justice standpoint anyways!

**Pull request closed without merging**